### PR TITLE
Update edgeql-parser

### PR DESCRIPTION
**Pull request by aljazerzen**
Updates usages of edgeql-parser library to match https://github.com/edgedb/edgedb/pull/5645

### CODE REVIEW

**Comment by tailhook** [comment]
I think it's better not have "python" in default features. Because it's there only for server. And other things, like wasm-lexer, CLI, and whatever tool that wants to work with EdgeQL (e.g. formatter) will never need it. I even think that feature should be called "edgedb-server-python" or "unstable-python" or "internal-python", to note that we aren't going to maintain that as a public API in any amount.

Is there a problem for building with a feature enabled for EdgeDB server?

File: src/migrations/create.rs [hunk 1]
```diff
@@ -788,11 +788,11 @@ pub async fn normal_migration(cli: &mut Connection, ctx: &Context,
 }
 
 fn add_newline_after_comment(value: &mut String) -> Result<(), anyhow::Error> {
-    let last_token = TokenStream::new(value).last()
+    let last_token = Tokenizer::new(value).last()
         .ok_or_else(|| bug::error("input should not be empty"))?
         .map_err(|e| bug::error(
-            format!("tokenizer failed on reparsing: {}", e)))?;
-    let token_end = last_token.end.offset as usize;
+            format!("tokenizer failed on reparsing: {}", e.message)))?;
```
Comment (tailhook) [inline src/migrations/create.rs:794]: What's wrong with `fmt::Display` of the error?

It's also probably should use `{:#}` in error format string, as that usually gives more detailed info on error (by a convention introduced by `anyhow`).

File: src/migrations/grammar.rs [hunk 2]
```diff
@@ -27,6 +28,8 @@ pub struct TokenMatch<'a> {
     phantom: PhantomData<&'a u8>,
 }
 
+pub struct TokenStream<'a>(pub Tokenizer<'a>);
```
Comment (tailhook) [inline src/migrations/grammar.rs:31]: Eventually should use chumsky here too. This doesn't have to be in PR. But I would appreciate a `// TODO` comment here

File: src/migrations/create.rs (hunk 1 above)
Comment (aljazerzen) [reply src/migrations/create.rs:794]: Oh, that's just a biproduct of changing errors from `combine::easy::Error` to our `edgeql_parser::Error`. I can implement Display for our error as well.

**Comment by aljazerzen** [comment]
> I think it's better not have "python" in default features of edgeql-parser.

Oh, yes that's right. I don't remember why I added it to the default features. Probably because I wanted to test the derive macro (that's enabled only for python feature) with simple `cargo check` instead of `cargo check --features=...`. I'll remove python from the default features.

**Comment by aljazerzen** [comment]
What's up with tests? Did I break something? They do have similar problem on my machine, so probably not.

**Comment by tailhook** [comment]
> What's up with tests? Did I break something?

No, edgedb sometimes fails to start on github actions. Retry usually works, but it's confusing what is the issue.

> They do have similar problem on my machine, so probably not.

Hm, can you take a look? All those `Lazy instance has previously been poisoned` are irrelevant. `valid port received: RecvError` this is where it can't get status from a server. I'm not sure how to figure out what's with server though, maybe increase logging or strace it.

**Comment by aljazerzen** [comment]
I didn't realize the test is running the edgedb-server, which I didn't have in my path. When I did, it was reporting this:

```
1. edb.buildmeta.MetadataError: could not find SHARED_DATA_DIR in EdgeDB distribution metadata
```
... which is probably caused by me having a development build.

When I added `edgedb-server` 2.16 into the path, this was the output:

<details>

```
---- migrations::unsafe_migrations stdout ----
thread 'migrations::unsafe_migrations' panicked at 'Unexpected failure.
code-1
stderr=```"edgedb error: Dev mode is not supported on EdgeDB 2.16+a95c8e1. Please upgrade.\n"```
command=`CLICOLOR="0" NO_COLOR="1" "/home/aljaz/EdgeDB/edgedb-cli/target/debug/edgedb" "--no-cli-update-check" "--admin" "--unix-path" "/tmp/tmpnfjqm10h" "--port" "45799" "--database=db_unsafe" "migrate" "--dev-mode" "--schema-dir=tests/migrations/db_unsafe/initial"`
code=1
stdout=""
stderr="edgedb error: Dev mode is not supported on EdgeDB 2.16+a95c8e1. Please upgrade.\n"
', /rustc/84c898d65adf2f39a5a98507f1fe0ce10a2b8dbc/library/core/src/ops/function.rs:250:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

---- migrations::dev_mode stdout ----
thread 'migrations::dev_mode' panicked at 'Unexpected failure.
code-1
stderr=```"edgedb error: Dev mode is not supported on EdgeDB 2.16+a95c8e1. Please upgrade.\n"```
command=`CLICOLOR="0" NO_COLOR="1" "/home/aljaz/EdgeDB/edgedb-cli/target/debug/edgedb" "--no-cli-update-check" "--admin" "--unix-path" "/tmp/tmpnfjqm10h" "--port" "45799" "--database=db4" "migrate" "--dev-mode" "--schema-dir=tests/migrations/db4/initial"`
code=1
stdout=""
stderr="edgedb error: Dev mode is not supported on EdgeDB 2.16+a95c8e1. Please upgrade.\n"
', /rustc/84c898d65adf2f39a5a98507f1fe0ce10a2b8dbc/library/core/src/ops/function.rs:250:5

---- non_interactive::database_create_wipe_drop stdout ----
------ create: create new database (STDOUT) -----

------ create: create new database (STDERR) -----
OK: CREATE DATABASE

------ add-data: add some data to the new database (STDOUT) -----
{"id": "7a25559e-0ad1-11ee-a1c9-b32166c31268"}

------ add-data: add some data to the new database (STDERR) -----
OK: CREATE TYPE

------ check-data: check that added data is still there (STDOUT) -----
{"id": "7a25559e-0ad1-11ee-a1c9-b32166c31268"}

------ check-data: check that added data is still there (STDERR) -----

------ wipe: wipe the data out (STDOUT) -----

------ wipe: wipe the data out (STDERR) -----
edgedb error: The `database wipe` command is only supported in EdgeDB >= 3.0
  Hint: Use `edgedb database drop`, `edgedb database create`

thread 'non_interactive::database_create_wipe_drop' panicked at 'Unexpected failure.
code-1
stderr=``````
edgedb error: The `database wipe` command is only supported in EdgeDB >= 3.0
  Hint: Use `edgedb database drop`, `edgedb database create`
```
```
command=`CLICOLOR="0" "/home/aljaz/EdgeDB/edgedb-cli/target/debug/edgedb" "--no-cli-update-check" "--admin" "--unix-path" "/tmp/tmpnfjqm10h" "--port" "45799" "database" "wipe" "--database=test_create_wipe_drop" "--non-interactive"`
wipe=`wipe the data out`
code=1
stdout=""
stderr=```
edgedb error: The `database wipe` command is only supported in EdgeDB >= 3.0
  Hint: Use `edgedb database drop`, `edgedb database create`
```

', /rustc/84c898d65adf2f39a5a98507f1fe0ce10a2b8dbc/library/core/src/ops/function.rs:250:5

---- migrations::modified2_interactive stdout ----
thread 'migrations::modified2_interactive' panicked at 'called `Result::unwrap()` on an `Err` value: EOF { expected: "\"Created tests/migrations/db1/modified2/migrations/00002.edgeql, id: m13wjyiog2dbum2ou32yp77eysbewews7vlv6rqqfswpyi2yd4s55a\"", got: "\r\n\u{1b}[?2004h\r\u{1b}[0K> \r\u{1b}[2Cy\u{1b}[?2004l\r\nCreated tests/migrations/db1/modified2/migrations/00002.edgeql, id: m1caxjxlggy5xv63isfp5oxdbucx35efhgevxdklvlcgjgpdus3j3q\r\n", exit_code: None }', tests/func/migrations.rs:536:70

---- migrations::modified1 stdout ----
thread 'migrations::modified1' panicked at 'Unexpected stdout, failed diff original var
├── original: m13wjyiog2dbum2ou32yp77eysbewews7vlv6rqqfswpyi2yd4s55a
├── diff: 
│   ---         orig
│   +++         var
│   @@ -1 +1 @@
│   -m13wjyiog2dbum2ou32yp77eysbewews7vlv6rqqfswpyi2yd4s55a
│   +m1caxjxlggy5xv63isfp5oxdbucx35efhgevxdklvlcgjgpdus3j3q
└── var as str: m1caxjxlggy5xv63isfp5oxdbucx35efhgevxdklvlcgjgpdus3j3q

command=`CLICOLOR="0" "/home/aljaz/EdgeDB/edgedb-cli/target/debug/edgedb" "--no-cli-update-check" "--admin" "--unix-path" "/tmp/tmpnfjqm10h" "--port" "45799" "migration" "log" "--from-fs" "--schema-dir=tests/migrations/db1/modified1" "--newest-first" "--limit=1"`
code=0
stdout="m1caxjxlggy5xv63isfp5oxdbucx35efhgevxdklvlcgjgpdus3j3q\n"
stderr=""
', /rustc/84c898d65adf2f39a5a98507f1fe0ce10a2b8dbc/library/core/src/ops/function.rs:250:5


failures:
    migrations::dev_mode
    migrations::modified1
    migrations::modified2_interactive
    migrations::unsafe_migrations
    non_interactive::database_create_wipe_drop
```

</details>

**Pull request merged**